AzureBatchQueue/MessageBatch.cs

```diff
@@ -0,0 +1,101 @@
+using System.Runtime.Serialization;
+using AzureBatchQueue.Utils;
+using Newtonsoft.Json;
+
+namespace AzureBatchQueue;
+
+public class MessageBatch<T>
+{
+    private readonly List<T> items = new();
+    public List<T> Items() => items;
+
+    private int MaxSizeInBytes { get; init; }
+    public const int AzureQueueMaxSizeInBytes = 49_000; // ~ 48 KB
+    public SerializerType SerializerType { get; init; }
+    private readonly IMessageBatchSerializer<T> serializer;
+    private int? batchSize;
+
+    public MessageBatch(IMessageBatchSerializer<T> serializer, int maxSizeInBytes = AzureQueueMaxSizeInBytes)
+    {
+        this.serializer = serializer;
+        MaxSizeInBytes = maxSizeInBytes;
+        SerializerType = serializer.GetSerializerType();
+    }
+
+    public MessageBatch(SerializerType serializerType, int maxSizeInBytes = AzureQueueMaxSizeInBytes)
+    {
+        serializer = GetSerializer(serializerType);
+        SerializerType = serializerType;
+        MaxSizeInBytes = maxSizeInBytes;
+    }
+
+    /// <summary>
+    /// Use this ctor if you are sure that all items will fit into the batch
+    /// </summary>
+    /// <param name="items"></param>
+    /// <param name="serializerType"></param>
+    /// <param name="maxSizeInBytes"></param>
+    public MessageBatch(List<T> items, SerializerType serializerType, int maxSizeInBytes = AzureQueueMaxSizeInBytes)
+        : this(serializerType, maxSizeInBytes) => this.items = items;
+
+    public bool TryAdd(T item)
+    {
+        items.Add(item);
+
+        var jsonSize = CalculateItemsSize();
```

Comment:
It seems like we perform heavy computation (compression) two times here — to calculate message size and then finally serialize all the items

Reply:
I've got the challenges with guessing a compressed size. 

What if we try our battle-tested approach - use blobs for large messages? Then, it would be up to clients which batch size to choose. The queue will use a blob if the message size is too large for a queue message. 

Should simplify logic a lot